src/main/scala/scala/xml/quote/internal/Liftables.scala

```diff
@@ -1,138 +1,152 @@
-package scala.xml.quote.internal
+package scala.xml.quote
+package internal
 
-import scala.reflect.macros.whitebox.Context
+import scala.language.implicitConversions
+import scala.reflect.macros.blackbox
+import scala.xml.quote.internal.{parsed => p}
 
-trait Liftables extends Nodes {
-  val c: Context
+trait Liftables {
+  val c: blackbox.Context
   import c.universe._
-  import c.universe.internal.reificationSupport.{SyntacticBlock => SynBlock}
 
-  val sx = q"_root_.scala.xml"
-  val sci = q"_root_.scala.collection.immutable"
-
-  def NullableLiftable[T](f: T => Tree): Liftable[T] = Liftable { v =>
-    if (v == null) q"null"
-    else f(v)
+  val args: List[Tree]
+
+  private val sx = q"_root_.scala.xml"
+
+  /** When we lift, we don't know if we are within an enclosing xml element
+    * which defines a scope. In some cases we will have to fix the scope.
+    *
+    * E.g:
+    * {{{
+    *   xml"""<a xmlns:pre="scope0">${ xml"<b/>" }</a>"""
+    * }}}
+    * Here the scope of `b` is `TopScope` but should be `scope0`
+    */
+  def fixScopes(tree: Tree): Tree = {
+    var isTopScope = true
+    c.internal.typingTransform(tree)((tree, api) => tree match {
+      case q"$sx.TopScope" if !isTopScope =>
+        api.typecheck(q"$$scope$$")
```

Comment:
@densh This doesn't work. It fails with `not found: value $scope$`

Here is the generated code for `xml"""<a xmlns:pre="scope0">${ xml"<b/>" }</a>"""`:
Before fixing scopes:
```scala
{
  val $tmpscope = _root_.scala.xml.NamespaceBinding("pre", "scope0", _root_.scala.xml.TopScope);
  {
    val $scope$ = $tmpscope;
    _root_.scala.xml.Elem(((null): String), "a", _root_.scala.xml.Null, $scope$, false, (({
      val $buf = new _root_.scala.xml.NodeBuffer();
      $buf.&+(((scala.xml.Elem.apply(((null): scala.Predef.String), "b", scala.xml.Null, scala.xml.TopScope, true, (({
        val $buf = new scala.xml.NodeBuffer();
        $buf
      }): _*))): scala.xml.Node));
      $buf
    }): _*))
  }
}
```
After fixing the scopes:
```scala
{
  val $tmpscope = _root_.scala.xml.NamespaceBinding("pre", "scope0", _root_.scala.xml.TopScope);
  {
    val $scope$ = $tmpscope;
    _root_.scala.xml.Elem(((null): String), "a", _root_.scala.xml.Null, $scope$, false, (({
      val $buf = new _root_.scala.xml.NodeBuffer();
      $buf.&+(((scala.xml.Elem.apply(((null): scala.Predef.String), "b", scala.xml.Null, $scope$, true, (({
        val $buf = new scala.xml.NodeBuffer();
        $buf
      }): _*))): scala.xml.Node));
      $buf
    }): _*))
  }
}
```
This compiles if I paste it in a repl. So it fails to typecheck during macro expansion. I don't know though...